net/portmapper/portmapper.go

```diff
@@ -60,15 +60,29 @@ type Client struct {
 	pcpSawTime  time.Time // time we last saw PCP was available
 	uPnPSawTime time.Time // time we last saw UPnP was available
 
-	localPort  uint16
-	pmpMapping *pmpMapping // non-nil if we have a PMP mapping
+	localPort uint16
+
+	mapping mapping // non-nil if we have a mapping
+}
+
+// Mapping represents a created port-mapping over some protocol.  It specifies a lease duration,
+// how to release the mapping, and whether the map is still valid.
+type mapping interface {
+	// isCurrent indicates whether a mapping is valid now, since most mappings have a lease.
+	isCurrent() bool
+	// release will attempt to unmap the established port mapping.
```

Comment:
"will attempt" and no return value implies this is async? Clarify. Does this block?